test/unit/modules/transport.js

```diff
@@ -13,119 +13,356 @@
+					__private.removePeer({}, 'Custom peer remove message');
+					expect(loggerStub.debug.called).to.be.true;
+					expect(loggerStub.debug.calledWith('Cannot remove empty peer')).to.be.true;
+					done();
+				});
 
-				it('should return false');
+				it('should return false', function (done) {
+					var result = __private.removePeer({}, 'Custom peer remove message');
+					expect(result).to.be.false;
+					done();
+				});
 			});
 
 			describe('when options.peer is defined', function () {
 
-				it('should call library.logger.debug');
+				var removeSpy, peerData;
+				var restoreRewiredDeps;
+
+				beforeEach(function (done) {
+					removeSpy = sinon.spy();
+					restoreRewiredDeps = TransportModule.__set__({
+						modules: {
+							peers: {
+								remove: removeSpy
+							}
+						}
+					});
+					peerData = {
+						ip: '127.0.0.1',
+						wsPort: 8000
+					};
+					done();
+				});
+
+				afterEach(function (done) {
+					restoreRewiredDeps();
+					done();
+				});
 
-				it('should call modules.peers.remove');
+				it('should call library.logger.debug', function (done) {
+					__private.removePeer({
+						peer: peerData
+					}, 'Custom peer remove message');
+					expect(loggerStub.debug.called).to.be.true;
+					done();
+				});
 
-				it('should call modules.peers.remove with options.peer');
+				it('should call modules.peers.remove with options.peer', function (done) {
+					__private.removePeer({
+						peer: peerData
+					}, 'Custom peer remove message');
+					expect(removeSpy.calledWith(peerData)).to.be.true;
+					done();
+				});
 			});
 		});
 
 		describe('receiveSignatures', function () {
 
-			it('should call library.schema.validate');
+			it('should call library.schema.validate with empty query.signatures', function (done) {
+				__private.receiveSignature = sinon.stub().callsArg(1);
+				library.schema.validate = sinon.stub().callsArg(2);
 
-			it('should call library.schema.validate with query');
+				__private.receiveSignatures({
+					signatures: []
+				}, function (err) {
+					expect(library.schema.validate.called).to.be.true;
+					done();
+				});
+			});
 
-			it('should call library.schema.validate with schema.signatures');
+			it('should call library.schema.validate with query.signatures', function (done) {
+				__private.receiveSignature = sinon.stub().callsArg(1);
+				library.schema.validate = sinon.stub().callsArg(2);
 
-			describe('when library.schema.validate fails', function () {
+				__private.receiveSignatures({
+					signatures: ['SIGNATURE123', 'SIGNATURE456']
+				}, function (err) {
+					expect(library.schema.validate.called).to.be.true;
+					done();
+				});
+			});
 
-				it('should call series callback with error = "Invalid signatures body"');
+			it('should call library.schema.validate with custom schema.signatures', function (done) {
+				var restoreRewiredDeps = TransportModule.__set__({
+					schema: {
```

Comment:
Why is the rest needed here?

Reply:
This one applies only to that single `it` test case; that's why I didn't put it in a before/after/beforeEach/afterEach block but I could probably put it in a before/after block. Now `restoreRewiredDeps()` is called on line 275.